IMDA-BTG/aiverify
# API connector to model server

## Description

Merging API Connector implementation

## Motivation and Context

Code changes is for the API Connector feature which enables user to test API model which is hosted on a remote server

## Type of Change


- feat: A new feature










## How to Test

- Spin up the mock remote API model server alongside AI Verify toolkit
- Entry point to API Connector Configuration UI is http://localhost:3000/assets/newModel
- Refer to https://imda-dsl.atlassian.net/wiki/spaces/ATT/pages/2161704961/API+Connector+Integration+Tests (Test Cases 1 to 9 and  13 to 15)
- For each testcase, create the config using the API Connector UI
- Then do the normal report generation workflow using the API Config as the Model

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x ]  I have tested the changes locally and verified that they work as expected.
- [x ]  I have added appropriate unit tests or functional tests for the changes made.
- [x ]  I have followed the project's coding conventions and style guidelines.
- [x ]  I have rebased my branch onto the latest commit of the main branch.
- [x ]  I have squashed or reorganized my commits into logical units.
- [x ]  I have added any necessary dependencies or packages to the project's build configuration.
- [x ]  I have performed a self-review of my own code.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
